client/components/App.jsx

```diff
@@ -61,15 +63,30 @@ export default class App extends PureComponent{
     super(props)
   }
 
-  static propTypes = {
-    params: PropTypes.shape ({
-      siteId: PropTypes.string
-    })
+  getRouter = () => {
+    let names = window.location.host.split('.')
+    names = names[0] === ('www') ? names.slice(1) : names // trim www
+
+    if (Config.subhosts.indexOf(names.slice(1).join('.')) >= 0)
+      // Sub host (club site)
+      return (
+        <Router history={browserHistory}>
+          <Route siteId={names[0]} component={SiteContainer} path="/" />
+          <Route siteId={names[0]} component={EditorContainer} path="/edit"/>
+        </Router>
+      )
+    else
+      // Main host (clubhub site)
+      return (
```

Comment:
For the sake of making the code easy to reason about -- especially if we add more routes -- I think we should have a single return statement.